### PR TITLE
feat: signup & theme-context

### PR DESCRIPTION
**Pull request by tusharkandpal**
To check:

- `theme-context.js`
- `auth-context.js`
- `Signup` page

**Note:** other jsx & style changes can be ignored.

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -12,6 +16,12 @@ a {
   color: black;
```
Comment (Kothalkarpriya) [inline src/App.css]: define colors into root at top or also you define all the colors into a single file and then access it into other files

File: src/App.css [hunk 2]
```diff
@@ -12,6 +16,12 @@ a {
...
 }
 
+/* THEME CHANGES */
+.dark, .dark a {
+  color: white;
+  background-color: #252525;
```
Comment (Kothalkarpriya) [inline src/App.css]: You need to improve it

File: src/context/auth-context.js [hunk 3]
```diff
@@ -12,15 +12,36 @@ const AuthProvider = ({ children }) => {
...
       authDispatch({ type: "FAILED", payload: { path: "Login" } });
+      setTimeout(() => {
+        authDispatch({ type: "RESET" });
+      }, 1500);
+    }
+  };
+
+  const signupHandler = async (email, password) => {
+    try {
+      const { data } = await axios.post("/api/auth/signup", {
```
Comment (Kothalkarpriya) [inline src/context/auth-context.js:32]: Good that you used async await here

**Review by Kothalkarpriya — APPROVED** [review]
Look into my suggestions and then you can merge

File: src/App.css (hunk 2 above)
Comment (tusharkandpal) [reply src/App.css]: you mean to create color variable ?

File: src/App.css (hunk 1 above)
Comment (tusharkandpal) [reply src/App.css]: sure!

**Review by ultronjs — APPROVED** [review]
The code looks fine. Just a suggestion same as Priya.Can work on CSS variable

File: src/App.css (hunk 2 above)
Comment (Kothalkarpriya) [reply src/App.css]: yes

**Pull request merged**